File: DSA/RATNA_KHANNA_590013141/day5question2.c
#include<stdio.h>
int main()
{
    int n;
    printf("enter number of elements in array\n");
    scanf("%d",&n);
    int nums[n];
    printf("enter the elements to the array\n");
    for(int i=0;i<n;i++)
    {
        scanf("%d",&nums[i]);
    }
         for (int i = 0; i < n-1; i++) 
     {
        for (int j = i + 1; j < n; j++) 
        {
            if (nums[i] > nums[j]) 
            {
                int temp = nums[i];
                nums[i] = nums[j];
                nums[j] = temp;
            }
        }
    }
      for(int i = 0; i <= n; i++) {
        if(nums[i] != i) {
            printf("Missing number is: %d\n", i);
            return 0;
        }
    }
return 0;
}

File: DSA/VenikaPanwar_590013755/Day_5_Question_2.java
import java.util.*;

public class Day_5_Question_2 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter array in format: {3 0 1}");
        String input = sc.nextLine();
        input = input.replace("{", "").replace("}", "").trim();
        String[] parts = input.split(" ");
        int[] nums = new int[parts.length];

        for (int i = 0; i < parts.length; i++) {
            nums[i] = Integer.parseInt(parts[i]);
        }

        int n = nums.length;
        int totalSum = n * (n + 1) / 2;
        int actualSum = 0;

        for (int i = 0; i < n; i++) {
            actualSum += nums[i];
        }

        int missing = totalSum - actualSum;
        System.out.println("Missing Number is: " + missing);
    }
}
